__tests__/integration/utils/test-utils-integration.js

```diff
@@ -916,6 +916,50 @@ export class TestUtils {
     return result;
   }
 
+  /**
+   * This method awaits a tx to be confirmed by a block
+   *
+   * It does not return any content, only delivers the code processing back to the caller at the
+   * desired time.
+   *
+   * @param {string} walletId ID of the wallet to get the transaction from
+   * @param {String} txId ID of the tx to wait confirmation
+   * @param {number | null | undefined} timeout
+   * @returns {Promise<void>}
+   */
+  static async waitTxConfirmed(walletId, txId, timeout) {
+    let timeoutHandler;
+    let timeoutReached = false;
+    if (timeout) {
+      // Timeout handler
+      timeoutHandler = setTimeout(() => {
+        timeoutReached = true;
+      }, timeout);
+    }
+
+    let number = await TestUtils.getTransactionConfirmationNumber(walletId, txId);
+
+    while (number < 1) {
+      if (timeoutReached) {
+        break;
+      }
+      // Tx not confirmed, wait 1s before trying again
+      await delay(1000);
+
+      number = await TestUtils.getTransactionConfirmationNumber(walletId, txId);
+    }
+
+    if (timeoutHandler) {
+      clearTimeout(timeoutHandler);
+    }
+
+    if (timeoutReached) {
+      // We must do the timeout handling like this because if I reject the promise directly
+      // inside the setTimeout block, this while block will continue running forever.
+      throw new Error(`Timeout of ${timeout}ms without confirming the tx with id ${txId}`);
+    }
+  }
```

Comment:
```suggestion
  static async waitTxConfirmed(walletId, txId, timeout) {
    const timeoutPromise = new Promise((_, reject) =>
      setTimeout(() =>
        reject(
          new Error(`Timeout of ${timeout}ms without confirming the tx with id ${txId}`)
        ),
        timeout
      )
    );

    const waitForConfirmationPromise = async () => {
      let number = await TestUtils.getTransactionConfirmationNumber(walletId, txId);

      while (number < 1) {
        if (timeoutReached) {
          break;
        }
        // Tx not confirmed, wait 1s before trying again
        await delay(1000);

        number = await TestUtils.getTransactionConfirmationNumber(walletId, txId);
      }

      // If the loop finished, the promise is resolved successfully, nothing to return.
    };

    return Promise.race([waitForConfirmationPromise, timeoutPromise]);
  }
```

(untested and just a suggestion for aesthetically pleasing code, Promise.race will either resolve successfully or throw, depending on what resolves or rejects first)

Reply:
I think the code looks more confusing with the recommendation. As discussed privately, I will keep as it is now.